zerver/migrations/0371_invalid_characters.py

```diff
@@ -0,0 +1,54 @@
+import unicodedata
+
+from django.db import connection, migrations, models
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def fix_topics(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Message = apps.get_model("zerver", "Message")
+    BATCH_SIZE = 10000
+    lower_bound = 0
+
+    max_id = Message.objects.aggregate(models.Max("id"))["id__max"]
+    if max_id is None:
+        # No Message rows.
+        return
+
+    while lower_bound < max_id:
+        with connection.cursor() as cursor:
+            cursor.execute(
+                "SELECT DISTINCT subject FROM zerver_message WHERE id > %s AND id <= %s",
+                [lower_bound, lower_bound + BATCH_SIZE],
+            )
+
+            results = cursor.fetchall()
+
+            topics = [r[0] for r in results]
+            for topic in topics:
+                fixed_topic = "".join(
+                    [
+                        character
+                        for character in topic
+                        if unicodedata.category(character) not in ["Cc", "Cs", "Cn"]
+                    ]
+                )
+                if fixed_topic == topic:
+                    continue
+                cursor.execute(
+                    "UPDATE zerver_message SET subject = %s WHERE subject = %s AND id > %s AND id <= %s",
+                    [fixed_topic, topic, lower_bound, lower_bound + BATCH_SIZE],
+                )
+            lower_bound += BATCH_SIZE
+
+
+class Migration(migrations.Migration):
+    atomic = False
+
+    dependencies = [
+        ("zerver", "0370_realm_enable_spectator_access"),
+    ]
+
+    operations = [
+        migrations.RunPython(fix_topics, reverse_code=migrations.RunPython.noop),
```

Comment:
Let's name the migration file `invalid_characters_in_topics`; it seems worth being a bit more specific.